packages/server/src/seeds/searchparameters.ts

```diff
@@ -4,19 +4,32 @@ import { getDatabasePool } from '../database';
 import { Repository, getSystemRepo } from '../fhir/repo';
 import { globalLogger } from '../logger';
 import { r4ProjectId } from '../seed';
+import { RebuildOptions, buildRebuildOptions } from './common';
 
 /**
  * Creates all SearchParameter resources.
+ * @param options - Optional options for how rebuild should be done.
  */
-export async function rebuildR4SearchParameters(): Promise<void> {
+export async function rebuildR4SearchParameters(options?: Partial<RebuildOptions>): Promise<void> {
+  const finalOptions = buildRebuildOptions(options);
   const client = getDatabasePool();
   await client.query('DELETE FROM "SearchParameter" WHERE "projectId" = $1', [r4ProjectId]);
 
   const systemRepo = getSystemRepo();
 
-  for (const filename of SEARCH_PARAMETER_BUNDLE_FILES) {
-    for (const entry of readJson(filename).entry as BundleEntry[]) {
-      await createParameter(systemRepo, entry.resource as SearchParameter);
+  if (finalOptions.parallel) {
+    const promises = [];
+    for (const filename of SEARCH_PARAMETER_BUNDLE_FILES) {
+      for (const entry of readJson(filename).entry as BundleEntry[]) {
+        promises.push(createParameter(systemRepo, entry.resource as SearchParameter));
+      }
+    }
+    await Promise.all(promises);
```

Comment:
Consider always building the array, and then using the `parallel` option in the last step

```ts
if (finalOptions.parallel) {
  await Promise.all(promises);
} else {
  for (const promise of promises) {
    await promise;
  }
}
```

That way we don't need to worry about the logic getting out of sync.

Reply:
Good point 👍

Reply:
Actually this would work. Filling the array in the serial case doesn't actually execute the promises serially, only waits for them serially. We have to create the promise in the loop where it is awaited in order to not unintentionally parallelize them

Reply:
I don't know if it's worth introducing a new dependency, but [p-limit](https://github.com/sindresorhus/p-limit) seems like a pretty elegant and simple solution to abstract away the max concurrency that we want to allow and avoid the `if/else` code paths.